report/src/pages/RunIndex.tsx

```diff
@@ -71,128 +72,94 @@ function RunIndex() {
     });
   }, [matchedRuns, filterSelections]);
 
-  if (!benchmarkRuns || isLoadingBenchmarkRuns) {
-    return <div>Loading...</div>;
-  }
-
-  return (
-    <div className="container mx-auto">
-      {/* Filter Interface */}
-      <div className="flex flex-wrap gap-4 pb-4 mb-4">
-        {Object.entries(filterOptions)
-          .sort((a, b) => a[0].localeCompare(b[0]))
-          .map(([key, availableValues]) => {
-            const currentValue = filterSelections[key] ?? "any";
-            return (
-              <div key={key}>
-                <div className="text-sm text-slate-500 mb-1">
-                  {camelToTitleCase(key)}
-                </div>
-                <select
-                  className="bg-white border border-slate-300 rounded-md px-3 py-1.5 text-sm focus:outline-none focus:ring-2 focus:ring-indigo-500 focus:border-indigo-500"
-                  value={String(currentValue)}
-                  onChange={(e) => {
-                    const newValue = e.target.value;
-                    setFilterSelections((prev) => {
-                      const newSelections = { ...prev };
-                      if (newValue === "any") {
-                        delete newSelections[key];
-                      } else {
-                        newSelections[key] = newValue;
-                      }
-                      return newSelections;
-                    });
-                  }}
-                >
-                  <option value="any">Any</option>
-                  {availableValues.map((val) => (
-                    <option value={String(val)} key={String(val)}>
-                      {formatLabel(String(val))}
-                    </option>
-                  ))}
-                </select>
-              </div>
-            );
-          })}
-      </div>
+  // Group matchedRuns by id and precompute group sections with diffKeyStart
+  const groupedSections = useMemo(() => {
+    // Group runs by id
+    const groups: Record<string, BenchmarkRun[]> = {};
```

Comment:
nit: could possibly simplify this using `_.groupBy` https://lodash.com/docs/#groupBy. For the grouping key, you can use `(group) => group.id ?? null`, which should result in a grouping with type `Record<string | null, BenchmarkRun[]>`